tensorflow_io/ignite/kernels/ggfs/ggfs.cc

```diff
@@ -0,0 +1,250 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+
+#include "tensorflow_io/ignite/kernels/ggfs/ggfs.h"
+#include "tensorflow/core/lib/core/errors.h"
+#include "tensorflow/core/lib/core/status.h"
+#include "tensorflow/core/lib/io/path.h"
+#include "tensorflow/core/platform/env.h"
+#include "tensorflow/core/platform/file_system.h"
+#include "tensorflow/core/platform/file_system_helper.h"
+#include "tensorflow_io/ignite/kernels/client/ignite_plain_client.h"
+#include "tensorflow_io/ignite/kernels/client/ignite_ssl_wrapper.h"
+#include "tensorflow_io/ignite/kernels/ggfs/ggfs_random_access_file.h"
+#include "tensorflow_io/ignite/kernels/ggfs/ggfs_writable_file.h"
+#include <queue>
+
+namespace tensorflow {
+
+GGFS::GGFS() {
+  LOG(INFO) << "Call GGFS::GGFS";
+
+  const char *env_host = std::getenv("IGNITE_HOST");
+  const char *env_port = std::getenv("IGNITE_PORT");
+  const char *env_username = std::getenv("IGNITE_USERNAME");
+  const char *env_password = std::getenv("IGNITE_PASSWORD");
+  const char *env_certfile = std::getenv("IGNITE_CERTFILE");
+  const char *env_keyfile = std::getenv("IGNITE_KEYFILE");
+  const char *env_cert_password = std::getenv("IGNITE_CERT_PASSWORD");
+
+  if (env_host)
+    host = string(env_host);
+
+  if (env_port && !strings::safe_strto32(env_port, &port)) {
+    LOG(ERROR) << "IGNITE_PORT environment variable is not a valid integer: "
+               << env_port;
+  }
+
+  if (env_username)
+    username = string(env_username);
+
+  if (env_password)
+    password = string(env_password);
+
+  if (env_certfile)
+    certfile = string(env_certfile);
+
+  if (env_keyfile)
+    keyfile = string(env_keyfile);
+
+  if (env_cert_password)
+    cert_password = string(env_cert_password);
+}
+
+GGFS::~GGFS() { LOG(INFO) << "Call GGFS::~GGFS"; }
+
+Status GGFS::NewRandomAccessFile(const string &file_name,
+                                 std::unique_ptr<RandomAccessFile> *result) {
+  LOG(INFO) << "Call GGFS::NewRandomAccessFile [file_name = " << file_name
+            << "]";
+
+  result->reset(new GGFSRandomAccessFile(
```

Comment:
It would be weird, wouldn't it? I guess the code that calls this method should be responsible for providing a not-null pointer. At least, in other places in such cases, we don't check it.